File: examples/PYTHON/send_attached_file.py
import urllib.parse
import urllib.request
import re
import json
import mimetypes
import os
import random

ENDPOINT = 'https://app.flapmail.net/api/submit.php'
# Gere uma chave Token e coloque aqui
FLAPMAIL_TOKEN = 'ms_replace_with_your_token'


def is_valid_email(email: str) -> bool:
    return re.match(r'^[^\s@]+@[^\s@]+\.[^\s@]+$', email) is not None

def build_multipart(fields: dict, file_field: str, file_path: str):
    boundary = '----FlapMailBoundaryPython'
    lines = []
    for key, value in fields.items():
        lines.append(f'--{boundary}')
        lines.append(f'Content-Disposition: form-data; name="{key}"')
        lines.append('')
        lines.append(str(value))
    filename = os.path.basename(file_path)
    mime = mimetypes.guess_type(filename)[0] or 'application/octet-stream'
    lines.append(f'--{boundary}')
    lines.append(f'Content-Disposition: form-data; name="{file_field}"; filename="{filename}"')
    lines.append(f'Content-Type: {mime}')
    lines.append('')
    body = '
'.join(lines).encode('utf-8') + b'
' + open(file_path, 'rb').read() + b'
' + f'--{boundary}--
'.encode('utf-8')
    return boundary, body

file_path = 'sample_attachment.txt'
if not os.path.exists(file_path):
    with open(file_path, 'w', encoding='utf-8') as handle:
        handle.write('Sample attachment for the FlapMail Python example.')
if os.path.getsize(file_path) > 3 * 1024 * 1024:
    raise SystemExit('The selected file exceeds the 3 MB limit.')
payload = {
    'name': 'Alex Johnson',
    'email': 'alex@example.com',
    'message': 'Please review the attached file.',
    'token': FLAPMAIL_TOKEN,
    # format can be 'html' or 'text'
    'format': 'html',
    'form_type': 'docs'
}
if not is_valid_email(payload['email']):
    raise SystemExit('Please enter a valid email address.')
boundary, body = build_multipart(payload, 'file', file_path)
request = urllib.request.Request(ENDPOINT, data=body, method='POST')
request.add_header('Content-Type', f'multipart/form-data; boundary={boundary}')
with urllib.request.urlopen(request, timeout=30) as response:
    print(response.read().decode('utf-8'))

File: examples/Node/send_attached_file.js
const https = require('https');
const querystring = require('querystring');
const fs = require('fs');
const path = require('path');

const FLAPMAIL_ENDPOINT_HOST = 'app.flapmail.net';
const FLAPMAIL_ENDPOINT_PATH = '/api/submit.php';
// Gere uma chave Token e coloque aqui
const FLAPMAIL_TOKEN = 'ms_replace_with_your_token';

function isValidEmail(email) {
  return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
}

function postUrlEncoded(payload) {
  const postData = querystring.stringify(payload);
  const options = { hostname: FLAPMAIL_ENDPOINT_HOST, path: FLAPMAIL_ENDPOINT_PATH, method: 'POST', headers: { 'Content-Type': 'application/x-www-form-urlencoded', 'Content-Length': Buffer.byteLength(postData) } };
  const req = https.request(options, (res) => { let body = ''; res.on('data', (chunk) => body += chunk); res.on('end', () => console.log(body)); });
  req.on('error', (error) => console.error('Request failed:', error.message));
  req.write(postData); req.end();
}

const payload = {
  name: 'Alex Johnson',
  email: 'alex@example.com',
  message: 'Please review the attached file.',
  token: FLAPMAIL_TOKEN,
  form_type: 'docs'
};
const filePath = path.join(__dirname, 'sample_attachment.txt');
if (!fs.existsSync(filePath)) {
  fs.writeFileSync(filePath, 'Sample file created for the FlapMail attachment example.');
}
if (!isValidEmail(payload.email)) {
  console.error('Please enter a valid email address.');
  process.exit(1);
}
const stats = fs.statSync(filePath);
if (stats.size > 3 * 1024 * 1024) {
  console.error('The selected file exceeds the 3 MB limit.');
  process.exit(1);
}
const boundary = '----FlapMailBoundary' + Date.now();
const fileName = path.basename(filePath);
const fileContent = fs.readFileSync(filePath);
const parts = [];
for (const [key, value] of Object.entries(payload)) {
  parts.push(Buffer.from(`--${boundary}
Content-Disposition: form-data; name="${key}"

${value}
`));
}
// format can be 'html' or 'text'
parts.push(Buffer.from(`--${boundary}
Content-Disposition: form-data; name="format"

html
`));
parts.push(Buffer.from(`--${boundary}
Content-Disposition: form-data; name="file"; filename="${fileName}"
Content-Type: application/octet-stream

`));
parts.push(fileContent);
parts.push(Buffer.from(`
--${boundary}--
`));
const body = Buffer.concat(parts);
const options = {
  hostname: FLAPMAIL_ENDPOINT_HOST,
  path: FLAPMAIL_ENDPOINT_PATH,
  method: 'POST',
  headers: {
    'Content-Type': `multipart/form-data; boundary=${boundary}`,
    'Content-Length': body.length
  }
};
const req = https.request(options, (res) => {
  let responseBody = '';
  res.on('data', (chunk) => responseBody += chunk);
  res.on('end', () => console.log(responseBody));
});
req.on('error', (error) => console.error('Request failed:', error.message));
req.write(body);
req.end();
